packages/text-input/src/TextInput.tsx

```diff
@@ -0,0 +1,331 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from '@leafygreen-ui/emotion';
+import Icon from '@leafygreen-ui/icon';
+import { uiColors } from '@leafygreen-ui/palette';
+import { useUsingKeyboardContext } from '@leafygreen-ui/leafygreen-provider';
+import { createDataProp } from '@leafygreen-ui/lib';
+
+export const State = {
+  None: 'none',
+  Valid: 'valid',
+  Error: 'error',
+} as const;
+
+export type State = typeof State[keyof typeof State];
+
+interface TextInputProps {
+  /**
+   * Text shown in bold above the input element.
+   */
+  label?: string;
+  /**
+   * Text that gives more detail about the requirements for the input.
+   */
+  description?: string;
+  /**
+   * Whether or not the field is optional.
+   * Default: false
+   */
+  optional?: boolean;
+  /**
+   * Whether or not the field is currently disabled.
+   * Default: false
+   */
+  disabled?: boolean;
+  /**
+   * Callback to be executed when the value of the input field changes.
+   */
+  onChange?: Function;
+  /**
+   * The placeholder text shown in the input field before the user begins typing.
+   */
+  placeholder?: string;
+  /**
+   * The message shown below the input field if the value is invalid.
+   */
+  errorMessage?: string;
+  /**
+   * The current state of the TextInput. This can be none, valid, or error.
+   */
+  state?: State;
```

Comment:
Now that I see this in code it seems it may be a confusing word when put against React state. But I don't really have a better alternative.

Reply:
`validity` is the only thing I can think of, but I don't think it's better.

I also think since we've moved away from class components, state is a bit less confusing since we just use the hook now.